store/transactions.js

```diff
@@ -1,41 +1,37 @@
 import Vue from 'vue'
 import axios from 'axios'
-import { wrapActionsWithResolvedNode } from './utils'
 
 export const state = () => ({
   transactions: {}
 })
 
 export const mutations = {
-  /**
-   * setTransaction object
-   * @param {Object} state
-   * @param {Object} transaction
-   */
-  setTransaction (state, transaction) {
-    Vue.set(state.transactions, transaction.hash, transaction)
+  setTransactions (state, transactions) {
+    for (let i = 0; i < transactions.length; i++) {
+      const transaction = transactions[i]
+      if (!state.transactions.hasOwnProperty(transaction.hash)) {
+        Vue.set(state.transactions, transaction.hash, transaction)
+      }
+    }
   }
 }
 
-export const actions = wrapActionsWithResolvedNode({
-  /**
-   *
-   * @param state
-   * @param commit
-   * @param {Object} rootGetters
-   * @param hash
-   * @return {Promise<*>}
-   */
-  async getTxByHash ({ state, commit, rootGetters: { node } }, hash) {
-    if (state.transactions[hash]) return state.transactions[hash]
-
-    const transaction = await node.tx(hash, true)
-    commit('setTransaction', transaction)
-
-    return transaction
+export const actions = {
+  getLatestTransactions: async function ({ state, rootState: { nodeUrl, height }, commit, dispatch }, payload) {
+    const page = payload.page
+    const maxTransactions = payload.numTransactions
+    try {
+      const transactions = await axios.get(nodeUrl + 'middleware/transactions/interval/1/' + height.toString() + '?limit=' + maxTransactions + '&page=' + page)
```

Comment:
why start generation is `1`?

Reply:
I need the last `x` transactions. Currently middleware have an endpoint to get transactions from generation `a` to generation `b`. If I make a request from gen `1` to gen `height`, and put a limit of `x`, I get the latest `x` transactions.

Reply:
@ror-shubham. ok for now then. but not the best approach to take. maybe we can create an endpoint for this in the middleware.